File: backend/app/models/conservation.py
# SPDX-FileCopyrightText: 2026 Dong Lab, Yale School of Medicine <https://donglab.org>
#
# SPDX-License-Identifier: Apache-2.0

from pydantic import BaseModel


class ConservationCell(BaseModel):
    gene_id: str
    symbol: str | None = None
    family: str | None = None
    species: str
    species_label: str | None = None
    perc_id: float | None = None
    perc_id_r1: float | None = None
    perc_pos: float | None = None
    dn_ds: float | None = None
    orthology_type: str | None = None
    ortholog_count: int = 0
    target_gene_id: str | None = None


class SpeciesNode(BaseModel):
    node_id: int
    parent_id: int | None = None
    branch_length: float
    species: str | None = None
    species_label: str | None = None
    taxon_id: int | None = None

File: frontend/src/types/conservation.ts
// SPDX-FileCopyrightText: 2026 Dong Lab, Yale School of Medicine <https://donglab.org>
//
// SPDX-License-Identifier: Apache-2.0

export interface ConservationCell {
  gene_id: string
  symbol: string | null
  family: string | null
  species: string
  species_label: string | null
  perc_id: number | null
  perc_id_r1: number | null
  perc_pos: number | null
  dn_ds: number | null
  orthology_type: string | null
  ortholog_count: number
  target_gene_id: string | null
}

export interface SpeciesNode {
  node_id: number
  parent_id: number | null
  branch_length: number
  species: string | null
  species_label: string | null
  taxon_id: number | null
}
